modules/pre-commit.nix

```diff
@@ -0,0 +1,49 @@
+{ config, lib, pkgs, ... }:
+
+let
+
+  inherit (lib)
+    mkIf
+    mkOption
+    types
+    ;
+
+  cfg = config.pre-commit;
+
+in
+{
+  options.pre-commit = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable pre-commit integration.
+
+        https://pre-commit.com/
+      '';
+    };
+
+    enableAutoInstall = mkOption {
```

Comment:
Why would someone set this to false?

Reply:
In case it doesn't work for them for whatever reason. It was supposed to be an escape hatch, but I think `activation.enableShellHook` is already good enough as an escape hatch. I think I wrote that one after this one. Will remove this one :+1: .